File: common50/binarySearch.py
def binarySearch(n,k):
    s = 0
    e = len(n)
    while(s<=e):
        mid = (s+e)//2
        if(n[mid]==k):
            print("Yes")
        elif(n[mid] < k):
            s = mid
        else:
            mid = e
n = [1,2,3,4,5,6]
l = len(n)
binarySearch(n,2)

File: common50/binarySearch.cpp
#include<bits/stdc++.h>
using namespace std;

void binary(vector<int> v, int k)
{
    int s = 0, e = v.size();
    while(s<= e)
    {
        int mid = (s+e)/2;

        if(v[mid] == k ) 
        {
            cout<<"Yes";
            break;
        }

        else if(v[mid] < k) s = mid; 

        else mid = e;
    }
}

int main()
{
    int a,k;
    vector<int>v ;
    cin>>a;
    int b;
    for(int i = 0; i< a; i++)
    {
        cin>>b;
        v.push_back(b);

    }
    cin>>k;

    binary(v,k);
    return 0;
}